### PR TITLE
mcuboot: Added support for NET core serial recovery when CONFIG_NRF_MULTI_IMAGE_UPDATE=n

### PR DESCRIPTION
**Pull request by nvlsianpu**
This PR makes possible serial recovery of the image of nRF5340 cpunet even that simultaneous cores upgrade is disabled.
In order to test this following mcumgr cli command should be used:
`./mcumgr image upload <build_dir_path>/zephyr/net_core_app_update.bin -e -n 3 -c serial_2`
where `serial_2` is the serial connection configuration.
 
ref. NCSDK-15698

Documentation PR: #8141 

``west build -d build/peripheral_uart_3 -b nrf5340dk_nrf5340_cpuapp nrf/samples/bluetooth/peripheral_uart/ -- -DOVERLAY_CONFIG=nrf5340dk_app_sr_net.conf -Dmcuboot_OVERLAY_CONFIG=../../../../nrf/samples/bluetooth/peripheral_uart/nrf5340dk_mcuboot_sr_net.conf```

### CODE REVIEW

**Review by sigvartmh — APPROVED** [review]

**Review by KAGA164 — CHANGES_REQUESTED** [review]

File: samples/bluetooth/peripheral_uart/boards/nrf5340dk_nrf5340_cpuapp.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CONFIG_MCUBOOT_IMAGE_VERSION="1.0.0+0"
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply samples/bluetooth/peripheral_uart/boards/nrf5340dk_nrf5340_cpuapp.conf:1]: @KAGA164 I opened separate PR for document serial recovery of the networking core image: #8141

[the remark this answers is not in the document]
Comment (nvlsianpu) [reply samples/bluetooth/peripheral_uart/boards/nrf5340dk_nrf5340_cpuapp.conf:1]: @KAGA164 I discussed this witch @grochu when you was OoO. It seam that best we can do is to add this as  optional configuration which can be included using Kconfig's OVERLAY_CONFIG mechanism and mentions this in the sample manual. General overview of the feature is under #8141

**Comment by nvlsianpu** [comment]
@KAGA164 Can you look here again?

File: samples/bluetooth/peripheral_uart/nrf5340dk_app_sr_net.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CONFIG_MCUBOOT_IMAGE_VERSION="1.0.0+0"
```
Comment (KAGA164) [inline samples/bluetooth/peripheral_uart/nrf5340dk_app_sr_net.conf:1]: Please add the license:
```
#
# Copyright (c) 2022 Nordic Semiconductor
#
# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
#
```

File: samples/bluetooth/peripheral_uart/nrf5340dk_mcuboot_sr_net.conf [hunk 3]
```diff
@@ -0,0 +1,32 @@
+CONFIG_MAIN_STACK_SIZE=10240
```
Comment (KAGA164) [inline samples/bluetooth/peripheral_uart/nrf5340dk_mcuboot_sr_net.conf:1]: Please add the license:
```
#
# Copyright (c) 2022 Nordic Semiconductor
#
# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
#
```

**Review by KAGA164 — COMMENTED** [review]
Looks good. Just please add license in configuration files

File: samples/bluetooth/peripheral_uart/README.rst [hunk 4]
```diff
@@ -21,6 +21,8 @@ The sample supports the following development kits:
    * The boards ``nrf52dk_nrf52810``, ``nrf52840dk_nrf52811``, and ``nrf52833dk_nrf52820`` only support the `Minimal sample variant`_.
    * When used with :ref:`zephyr:thingy53_nrf5340`, the sample supports the MCUboot bootloader with serial recovery and SMP DFU over Bluetooth.
      Thingy:53 has no built-in SEGGER chip, so the UART 0 peripheral is not gated to a USB CDC virtual serial port.
+   * When used with :ref:`zephyr:nrf5340dk_nrf5340`, the sample might supports the MCUboot bootloader with serial recovery of the networking
+     core image.
```
Comment (peknis) [inline samples/bluetooth/peripheral_uart/README.rst:25]:
```suggestion
   * When used with :ref:`zephyr:nrf5340dk_nrf5340`, the sample might support the MCUboot bootloader with serial recovery of the networking core image.
```

File: samples/bluetooth/peripheral_uart/README.rst [hunk 5]
```diff
@@ -94,6 +96,13 @@ Async adapter experimental module
    The module requires the :kconfig:option:`CONFIG_BT_NUS_UART_ASYNC_ADAPTER` to be set to ``y``.
    For more information about the adapter, see the :file:`uart_async_adapter` source files available in the :file:`peripheral_uart/src` directory.
 
+MCUboot with serial recovery of the networking core image
+=========================================================
+
+For the ``nrf5340dk_nrf5340_cpuapp`` it is possible to enable serial recovery of the network core while multi-image update is not enabled in the MCUboot.
```
Comment (peknis) [inline samples/bluetooth/peripheral_uart/README.rst:102]:
```suggestion
For the ``nrf5340dk_nrf5340_cpuapp``, it is possible to enable serial recovery of the network core while multi-image update is not enabled in the MCUboot.
```

File: samples/bluetooth/peripheral_uart/README.rst [hunk 6]
```diff
@@ -94,6 +96,13 @@ Async adapter experimental module
...
+See :ref:`peripheral_uart_sample_activating_variants` for details about how to build the sample with this feature using the :file:`nrf5340dk_app_sr_net.conf`
+and the :file:`nrf5340dk_mcuboot_sr_net.conf`.
```
Comment (peknis) [inline samples/bluetooth/peripheral_uart/README.rst:104]:
```suggestion
See :ref:`peripheral_uart_sample_activating_variants` for details on how to build the sample with this feature using the :file:`nrf5340dk_app_sr_net.conf` and :file:`nrf5340dk_mcuboot_sr_net.conf` files.
```

File: samples/bluetooth/peripheral_uart/README.rst [hunk 7]
```diff
@@ -144,6 +153,8 @@ To activate the optional extensions supported by this sample, modify :makevar:`O
 * For the minimal build variant, set :file:`prj_minimal.conf`.
 * For the USB CDC ACM extension, set :file:`prj_cdc.conf`.
   Additionally, you need to set :makevar:`DTC_OVERLAY_FILE` to :file:`usb.overlay`.
+* For the MCUboot with serial recovery of the networking core image feature, set :file:`nrf5340dk_app_sr_net.conf`.
+  Additinally, you need to set :makevar:`mcuboot_OVERLAY_CONFIG` to :file:`nrf5340dk_mcuboot_sr_net.conf`.
```
Comment (peknis) [inline samples/bluetooth/peripheral_uart/README.rst:157]:
```suggestion
* For the MCUboot with serial recovery of the networking core image feature, set the :file:`nrf5340dk_app_sr_net.conf` file.
  You also need to set the :makevar:`mcuboot_OVERLAY_CONFIG` variant to :file:`nrf5340dk_mcuboot_sr_net.conf`.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by KAGA164 — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@peknis I applied your suggestions.

**Review by peknis — APPROVED** [review]
Would this addition be worth a changelog entry?

**Comment by nvlsianpu** [comment]
@peknis 

> Would this addition be worth a changelog entry?

Yes, Of course.

File: west.yml [hunk 8]
```diff
@@ -103,7 +103,7 @@ manifest:
       revision: 76bd09efa5424a4039a234a7244fe32fa495d9cf
     - name: mcuboot
       repo-path: sdk-mcuboot
-      revision: v1.9.99-ncs2-rc1
+      revision: pull/206/head
```
Comment (rlubos) [inline west.yml:106]: Please update SHA: `8b3d00abb132de5f5c46310f88c2208ddf6270d4`

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
CI passed on reruns.

**Pull request merged**